src/components/subnavigation/navbar.tsx

```diff
@@ -0,0 +1,44 @@
+import NavButton from './nav-button';
+import DeveloperTopics from './developer-topics';
+import navButtons from '../../data/navbuttons';
+import React, { useState } from 'react';
+import { NavbarWrapper } from '../../styled/nav-bar';
+import DeveloperTopicsMenu from './developer-topics-menu';
+
+interface ContextProps {
+    open: boolean;
+    setIsOpen: (open: boolean) => void;
+}
+
+export const Context = React.createContext<ContextProps>({
+    open: false,
+    setIsOpen: (open: boolean) => {
+        /**/
+    },
+});
```

Comment:
Since it seems like we are only passing these values down 1 level (to `DeveloperTopic` and `DeveloperTopicsMenu`), what is the benefit of using Context here rather than just passing them as props?

Reply:
Correct. This goes back to your other comment "developer topics menu also closes when we click outside of the menu". If I have to handle that state change, can I still go with Prop drilling.

Reply:
Ah yeah that would sort of complicate it, I think maybe something could be done with [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) to help with this like is shown in [this post](https://stackoverflow.com/questions/32553158/detect-click-outside-react-component), but I think we can just cross that bridge when we come to it.